src/api/server.rkt

```diff
@@ -50,6 +61,45 @@
                     #:timeline-disabled? [timeline-disabled? #f])
   (herbie-command command test seed pcontext profile? timeline-disabled?))
 
+;; TODO move these side worker/manager
+(define (check-and-send path job-id page)
+  (define result-hash (get-results-for job-id))
+  (cond
+    [(set-member? (all-pages result-hash) page)
+     ;; Write page contents to disk
+     (when (*demo-output*)
+       (write-results-to-disk result-hash path))
+     (response 200
+               #"OK"
+               (current-seconds)
+               #"text"
+               (list (header #"X-Job-Count" (string->bytes/utf-8 (~a (job-count)))))
+               (λ (out)
+                 (with-handlers ([exn:fail? (page-error-handler result-hash page out)])
+                   (make-page page out result-hash (*demo-output*) #f))))]
+    [else #f]))
+
+(define (write-results-to-disk result-hash path)
```

Comment:
Probably reasonable to keep this in `server.rkt`, I could kind of go either way but having the caching stuff live here, which also involves writing things to disk, seems correct.